doc/themes/scikit-learn-modern/layout.html

```diff
@@ -9,8 +9,9 @@
 {%- set lang_attr = 'en' %}
 
 <!DOCTYPE html>
-<!--[if IE 8]><html class="no-js lt-ie9" lang="{{ lang_attr }}" > <![endif]-->
-<!--[if gt IE 8]><!--> <html class="no-js" lang="{{ lang_attr }}" > <!--<![endif]-->
+<!-- data-theme below is forced to be "light" but should be changed if we use pydata-theme-sphinx in the future -->
+<!--[if IE 8]><html class="no-js lt-ie9" lang="{{ lang_attr }}" data-theme="light"> <![endif]-->
+<!--[if gt IE 8]><!--> <html class="no-js" lang="{{ lang_attr }}" data-theme="light"> <!--<![endif]-->
```

Comment:
The downstream library probably use the `pydata-sphinx-theme` and therefore define this css selector :) We will need to come back and use the `pydata-sphinx-theme` at some point.

Reply:
You're convinced that the prevalence of the pydata-sphinx-theme is so high?

Reply:
Oh I see what you mean. Let me check where we can add the supported CSS variable.

Reply:
I had a quick look, and some do, but many use the readthedocs theme, and some do include sphinx-gallery in them. One example being mapie:
https://mapie.readthedocs.io/en/latest/examples_calibration/index.html

Reply:
> You're convinced that the prevalence of the pydata-sphinx-theme is so high?

I mean that the trick implemented here is only for the scikit-learn website that only use `sphinx-gallery` without the theme. In short:

- we want to use `sg-color-background` defined by sphinx-gallery
- it only exist if the CSS selector `data-theme` exist and this is indeed defined by `pydata-sphinx-theme`
- thus we need to redefine the `data-theme` in HTML because we don't use `pydata-sphinx-theme`

Something that we could do is to define our own scikit-learn css variable and document it for downstream library if they want to overwrite all others.

Reply:
> and some do include sphinx-gallery in them

but at the end I am wondering if this is not more of a sphinx-gallery bug since they should have a fallback for the CSS variable if someone does not use `pydata-sphinx-theme`

Reply:
> but at the end I am wondering if this is not more of a sphinx-gallery bug

I'm not following, but if you think that something can be done in sphinx-gallery to improve the situation, I think that it might be useful to explain it on the sphinx-gallery tracker (and ideally maybe contribute a PR)

Reply:
I will report upstream to see what is best.

Reply:
I open https://github.com/sphinx-gallery/sphinx-gallery/issues/1221.